src/NuGet.Core/NuGet.Build.Tasks/GetRestoreSolutionProjectsTask.cs

```
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace NuGet.Build.Tasks
+{
+    public class GetRestoreSolutionProjectsTask : Task
+    {
+        private const string MetaProjExtension = ".metaproj";
+
+        /// <summary>
+        /// Solution project references.
+        /// </summary>
+        [Required]
+        public ITaskItem[] ProjectReferences { get; set; }
+
+        /// <summary>
+        /// Root project path used for resolving the absolute path.
+        /// </summary>
+        [Required]
+        public string SolutionFilePath { get; set; }
+
+        /// <summary>
+        /// Output items
+        /// </summary>
+        [Output]
+        public ITaskItem[] OutputProjectReferences { get; set; }
+
+        public override bool Execute()
+        {
+            // Log inputs
+            var log = new MSBuildLogger(Log);
+            log.LogDebug($"(in) ProjectReferences '{string.Join(";", ProjectReferences.Select(p => p.ItemSpec))}'");
+            log.LogDebug($"(in) SolutionFilePath '{SolutionFilePath}'");
+
+            var entries = new List<ITaskItem>();
+
+            // Filter obvious duplicates without considering OS case sensitivity.
+            // This will be filtered further when creating the spec.
+            var seen = new HashSet<string>(StringComparer.Ordinal);
+            var parentDirectory = Path.GetDirectoryName(SolutionFilePath);
+
+            foreach (var project in ProjectReferences)
+            {
+                if (string.IsNullOrEmpty(project.ItemSpec))
```

Should it fail or warn on invalid value?

---

It should not be, but if it is the error/warning would be very unhelpful since it would not refer to anything.